### DISCUSSION TITLE
Get a branch by using a commit

### DISCUSSION THREAD

**Discussion by CristianCojocari** [gitpython-developers/GitPython#1402] [Q&A]
The code in question is here:

```
repository = repo.Repo(https://....)
repo.git.checkout('dev')
specifiedCommit = repo.commit('ACommitFromMasterBranch')
```

Is there a way to either check whether specifiedCommit is a commit in the dev branch, or restrict somehow the commits that can be accessed to the active branch?

Currently the code above outputs the actual commit from the master branch and I couldn't find any way to access it's branch to check whether it's dev or not.

**Comment by Byron** [comment]
GitPython doesn't have a fast way of determining if a commit is in the ancestry of a branch, as one would have to traverse it to know which can be slow.
One can use `git branch --contains` or `repo.git.branch('dev', contains='commit')` and parse the result.

**Reply by CristianCojocari** [reply]
> GitPython doesn't have a fast way of determining if a commit is in the ancestry of a branch, as one would have to traverse it to know which can be slow. One can use `git branch --contains` or `repo.git.branch('dev', contains='commit')` and parse the result.

This solution is perfect for me, thanks a lot!
